protocol/bc/asset_witness.go

```diff
@@ -0,0 +1,76 @@
+package bc
+
+import (
+	"fmt"
+	"io"
+
+	"chain/crypto/sha3pool"
+	"chain/encoding/blockchain"
+	"chain/errors"
+)
+
+type AssetWitness struct {
```

Comment:
`IssuanceWitness` got turned into `AssetWitness` somewhere between brain and keyboard, and stuck unnoticed somehow. Now fixed. Also added some Godoc.